systems/plants/setupLCPmex.cpp

```diff
@@ -0,0 +1,275 @@
+    
+    MatrixXd D_possible(mC*nC, nq);
+    for (int i = 0; i < mC ; i++) {
+      Map<MatrixXd> D_i(mxGetPr(mxGetCell(prhs[6], i)), numContactPairs , nq);
+      MatrixXd D_i_possible, D_i_exclude;
+      filterByIndices(possible_contact_indices, D_i, D_i_possible);
+      D_possible.block(nC*i, 0, nC, nq) = D_i_possible;
+    }
+
+    MatrixXd J(LCP_size, nq);
+    J << JL_possible, JP, n_possible, D_possible, MatrixXd::Zero(nC, nq);
+
+    Mqdn = Hinv*J.transpose();
+    
+    //solve LCP problem 
+    //TODO: call fastQP first
+    //TODO: call path from C++ (currently only 32-bit C libraries available)
+    mxArray* mxM = mxCreateDoubleMatrix(LCP_size, LCP_size, mxREAL);
+    mxArray* mxw = mxCreateDoubleMatrix(LCP_size, 1, mxREAL);
+    mxArray* mxlb = mxCreateDoubleMatrix(LCP_size, 1, mxREAL);
+    mxArray* mxub = mxCreateDoubleMatrix(LCP_size, 1, mxREAL);  
+    Map<VectorXd> lb(mxGetPr(mxlb), LCP_size);
+    Map<VectorXd> ub(mxGetPr(mxub), LCP_size);
+    lb << VectorXd::Zero(nL),
+          -1e20*VectorXd::Ones(nP),
+          VectorXd::Zero(nC+mC*nC+nC);
+    ub = 1e20*VectorXd::Ones(LCP_size);
+    
+    Map<MatrixXd> M(mxGetPr(mxM),LCP_size, LCP_size);
+    Map<VectorXd> w(mxGetPr(mxw), LCP_size);
+
+    //build LCP matrix
+    M << h*JL_possible*Mqdn,
+         h*JP*Mqdn,
+         h*n_possible*Mqdn,
+         D_possible*Mqdn,
+         MatrixXd::Zero(nC, LCP_size);
+    
+
+    if (nC > 0) {
+      for (int i=0; i < mC ; i++) {
+        M.block(nL+nP+nC+nC*i, nL+nP+nC+mC*nC, nC, nC) = MatrixXd::Identity(nC, nC);
+        M.block(nL+nP+nC+mC*nC, nL+nP+nC+nC*i, nC, nC) = -1.0*MatrixXd::Identity(nC, nC);
+      }
+      double mu = 1.0; //TODO: pull this from contactConstraints
+      M.block(nL+nP+nC+mC*nC, nL+nP, nC, nC) = mu*MatrixXd::Identity(nC, nC);
+    }
+
+    //build LCP vector
+    w << phiL_possible + h*JL_possible*wqdn,
+         phiP + h*JP*wqdn,
+         phiC_possible + h*n_possible*wqdn,
+         D_possible*wqdn,
+         VectorXd::Zero(nC);
+
+    mxArray *lhs[1];
+    mxArray *rhs[] = {mxM, mxw, mxlb, mxub};
+    
+    //call solver
+    mexCallMATLAB(1, lhs, 4, rhs, "pathlcp");
+    Map<VectorXd> z_path(mxGetPr(lhs[0]), LCP_size);
+    z = z_path;
+
+    //clean up
+    mxDestroyArray(lhs[0]);
+    mxDestroyArray(mxlb);
+    mxDestroyArray(mxub);
+    mxDestroyArray(mxM);
+    mxDestroyArray(mxw);
+    
+    VectorXd qdn = Mqdn*z + wqdn;
+
+    vector<int> impossible_contact_indices, impossible_limit_indices;
+    getInclusionIndices(possible_contact, impossible_contact_indices, false);
+    getInclusionIndices(possible_jointlimit, impossible_limit_indices, false);
+
+    vector<bool> penetrating_joints, penetrating_contacts;
+    getThresholdInclusion(phiL_check + h*JL_check*qdn, 0.0, penetrating_joints);
+    getThresholdInclusion(phiC_check + h*n_check*qdn, 0.0, penetrating_contacts);
+
+    const int pJoints = getNumTrue(penetrating_joints);
+    const int pContacts = getNumTrue(penetrating_contacts);
```

Comment:
could be mistaken for pointer. Maybe num_penetrating_joints, num_contacts?
Also shouldn't use camelcase.